bin/library.ps1

```diff
@@ -1328,118 +1498,2224 @@ namespace sqlcollective.dbatools
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="days"></param>
+            /// <param name="hours"></param>
+            /// <param name="minutes"></param>
+            /// <param name="seconds"></param>
+            public DbaTimeSpan(int days, int hours, int minutes, int seconds)
+            {
+                _timespan = new TimeSpan(days, hours, minutes, seconds);
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="days"></param>
+            /// <param name="hours"></param>
+            /// <param name="minutes"></param>
+            /// <param name="seconds"></param>
+            /// <param name="milliseconds"></param>
+            public DbaTimeSpan(int days, int hours, int minutes, int seconds, int milliseconds)
+            {
+                _timespan = new TimeSpan(days, hours, minutes, seconds, milliseconds);
+            }
+            #endregion Constructors
+
+            #region Methods
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="ts"></param>
+            /// <returns></returns>
+            public TimeSpan Add(TimeSpan ts)
+            {
+                return _timespan.Add(ts);
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="value"></param>
+            /// <returns></returns>
+            public int CompareTo(System.Object value)
+            {
+                return _timespan.CompareTo(value);
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="value"></param>
+            /// <returns></returns>
+            public int CompareTo(TimeSpan value)
+            {
+                return _timespan.CompareTo(value);
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="value"></param>
+            /// <returns></returns>
+            public int CompareTo(DbaTimeSpan value)
+            {
+                return _timespan.CompareTo(value.GetBaseObject());
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <returns></returns>
+            public TimeSpan Duration()
+            {
+                return _timespan.Duration();
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="value"></param>
+            /// <returns></returns>
+            public override bool Equals(System.Object value)
+            {
+                return _timespan.Equals(value);
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="obj"></param>
+            /// <returns></returns>
+            public bool Equals(TimeSpan obj)
+            {
+                re
```

Comment:
Applied your update - stylistically a lot superior to my version, so I'm all in favor-